cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/BaseStorage.java

```diff
@@ -41,4 +43,37 @@ public Map<String, String> getProperties() {
         .map(HashMap::new)
         .orElseGet(HashMap::new);
   }
+
+  /**
+   * Allocates Table Space for the storage.
+   *
+   * <p>Default tableLocation looks like: {endpoint}/{rootPrefix}/{databaseId}/{tableId}-{tableUUID}
+   *
+   * @return the table location where the table data should be stored
+   */
+  @Override
+  public String allocateTableSpace(
+      String databaseId,
+      String tableId,
+      String tableUUID,
+      String tableCreator,
+      boolean skipProvisioning) {
```

Comment:
what is skipProvisioning? javadoc missing too for params

Reply:
skipProvisioning is a way to get TableLocation without the allocation logic, its code will look like

```
String allocateTableSpace() {
   if (!skipProvisioning) {
        allocateBucket()
        setPermission()
        createSubdirectories()
   }
   return tableLocation;
}
```

We can use `allocateTableSpace(skipprovisioning=true)` to quickly get the table location. The use case for this is CTAS, when we try to match tableLocation from the `putSnapshot` call without rerunning the logic.

Reply:
So the name of the function is allocate - table - location and it has a flag that says skip provisioning in which case the function acts like a get - table - location? Something is off. Is there another way to support CTAS?

Reply:
@ctrezzo  had a similar feedback of splitting the methods [into allocateTableStorage, getTableLocation](https://github.com/linkedin/openhouse/pull/119#discussion_r1630366741). I can adopt this feedback as its common.  My reasoning was:
> it's possible to split into two methods: allocateStorage, getTableLocation. The reason I chose a single method was to avoid the split and centralize the whole logic into a single method. This way we avoid getTableLocation as a utility method and the usage of allocateStorage is limited during tableCreation time (either CT or CTAS). Let me know if you feel strongly and we can discuss more.

Let me know what you think. I'm not sure if we can support CTAS in a different way in the short term.

Reply:
So we want the createTableStartTransaction in CTAS to call this method with all the below steps.

- allocateBucket() 
- setPermission() 
- createSubdirectories()

We do not want the createTableCommitTransaction in CTAS to not call this method allocateTableSpace because it was already called as part of createTableStartTransaction.

Shouldn't the caller check, if already_staged = true which would be the case for createTableCommitTransaction do NOT call allocateTableSpace?

Why would we need getTableLocation as the additional method here?

Reply:
Thanks for the comments Sumedh, its clear the parameter is adding confusion right now, we'll introduce it back when we implement CTAS. 

to answer your questions,

- createTableStartTransaction in CTAS will not call this method to allocateBucket() etc. instead it will call it toGetTablePath so that it can do comparison like `metadata_json_path.isPrefix(allocateTableStorage(skipAllocation=true))`

Reply:
Yes, let's discuss it when we reach use-case where we need to support stage creation of table directory at the start of the transaction. Otherwise lgtm, do wait for @ctrezzo to approve this as he has a lot of good comments.